api/src/lib.rs

```diff
@@ -322,14 +322,24 @@ where
         secret: String,
     }
 
-    let body: VerifyVerifyBody = req.body_json().await.map_err(server_err)?;
+    let body: VerifyVerifyBody = req.body_json().await.map_err(bad_request)?;
 
     let db = &req.state().db;
     let id = &req.ext::<VaultId>().unwrap().0;
 
-    verify_token::verify(db, id, &body.secret).await.map_err(server_err)?;
-
-    Ok(StatusCode::Ok)
+    match verify_token::verify(db, id, &body.secret).await {
```

Comment:
this is fine. just for ref, I think there's a way to have our custom errors implement into<tide::Request> or something so that we only have to write this mapping code in one spot. but not pressing right now